haskell-src/Concordium/ID/IdentityProvider.hs

```diff
@@ -38,6 +39,57 @@ foreign import ccall unsafe "ip_info_url" ipUrlFFI :: Ptr IpInfo -> Ptr CSize ->
 foreign import ccall unsafe "ip_info_description" ipDescriptionFFI :: Ptr IpInfo -> Ptr CSize -> IO (Ptr Word8)
 foreign import ccall unsafe "ip_info_verify_key" ipVerifyKeyFFI :: Ptr IpInfo -> Ptr CSize -> IO (Ptr Word8)
 foreign import ccall unsafe "ip_info_cdi_verify_key" ipCdiVerifyKeyFFI :: Ptr IpInfo -> Ptr CSize -> IO (Ptr Word8)
+foreign import ccall unsafe "ip_info_create" createIpInfoFFI ::
+    -- The identity of the IP, for instance returned by `ipIdentityFFI`.
+    IdentityProviderIdentity ->
+    -- Pointer to a byte array which is the binary representation of a
+    -- `ed25519_dalek::PublicKey` Rust-instance, for instance one returned
+    -- by ipVerifyKeyFFI, and its length.
+    Ptr Word8 -> CSize ->
+    -- Pointer to a byte array which is the binary representation of a
+    -- `ps_sig::PublicKey<Bls12>` Rust-instance, for instance one returned
+    -- by ipCdiVerifyKeyFFI, and its length.
+    Ptr Word8 -> CSize ->
+    -- Pointer to a byte array which is the binary representation of an
+    -- utf8 encoded string, for instance one returned by ipNameFFI, and
+    -- its length.
+    Ptr Word8 -> CSize ->
+    -- Pointer to a byte array which is the binary representation of an
+    -- utf8 encoded string, for instance one returned by ipUrlFFI, and
+    -- its length.
```

Comment:
As I wrote before, it is not relevant where the pointer is coming from. And in any case in the place this is used the byte array is not coming from "ipUrlFFI". So please remove these "for instance" since they do not bring value.

What is relevant is that the pointer is of the right size, and the data is valid utf8.